keps/sig-scheduling/1258-default-pod-topology-spread/README.md

```diff
@@ -259,13 +272,200 @@ To ensure this feature to be rolled out in high quality. Following tests are man
+
+    - Will enabling / disabling the feature require downtime or reprovisioning
+      of a node? (Do not assume `Dynamic Kubelet Config` feature is enabled).
+
+      No
+
+* **Does enabling the feature change any default behavior?**
+
+  Yes. Users might experience more spreading of Pods among Nodes and Zones in certain topology distributions.
+  In particular, this will be more noticeable in clusters with more than 100 nodes.
+
+  The specific configuration was chosen to produce a behavior that closely resembles the `SelectorSpread` plugin.
+  See [this PR description](https://github.com/kubernetes/kubernetes/pull/91793) for simulation data.
+
+* **Can the feature be disabled once it has been enabled (i.e. can we roll back
+  the enablement)?**
+
+  Yes. Once disabled, only scheduling of new Pods will be affected.
+
+* **What happens if we reenable the feature if it was previously rolled back?**
+
+  Only scheduling of new Pods is affected.
+
+* **Are there any tests for feature enablement/disablement?**
+
+  There are unit tests in `pkg/scheduler/algorithmprovider/registry_test.go` that exercise the configuration of `kube-scheduler` with the plugins that correspond to the Feature Gate enablement.
+
+### Rollout, Upgrade and Rollback Planning
+
+* **How can a rollout fail? Can it impact already running workloads?**
+
+  Running workloads are not affected by `kube-scheduler`.
+
+* **What specific metrics should inform a rollback?**
+
+  Primarily scheduling latency metrics, such as `framework_extension_point_duration_seconds`, `scheduling_algorithm_duration_seconds`
+  and `e2e_scheduling_duration_seconds`, when they have increased significantly.
+
+  Since spreading is affected, node utilization might change.
+  Utilization metrics can be queried in the `/metrics/resource` endpoint exposed by kubelet.
+
+* **Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?**
+
+  N/A.
+
+* **Is the rollout accompanied by any deprecations and/or removals of features, APIs, 
+  fields of API types, flags, etc.?**
+
+  TBD for GA.
+
+### Monitoring Requirements
+
+* **How can an operator determine if the feature is in use by workloads?**
+
+  All Pods are affected, unless they have explicit spreading constraints (.spec.topologySpreadConstraints).
+
+* **What are the SLIs (Service Level Indicators) an operator can use to determine 
+  the health of the service?**
+
+  - [x] Metrics
+    - Metric name: `framework_extension_point_duration_seconds` with label `extension_point` values `PreScore` and/or `Score`.
+    - [Optional] Aggregation method:
+    - Components exposing the metric: `kube_scheduler`.
+  - [ ] Other (treat as last resort)
+    - Details:
+
+* **What are the reasonable SLOs (Service Level Objectives) for the above SLIs?**
+
+  For 100 nodes, with a 4-core master:
+
+  - Latency for PreScore less than 15ms for 99% percentile.
+  - Latency for Score less than 50ms for 99% percentile.
```

Comment:
Sounds like quite a lot, isn't it?

Reply:
I agree, we don't need to break them down I guess, but the total overhead of scoring (pre and actual scoring) should be  15ms.

Reply:
I used the numbers from our current performance dashboards. I guess we can include 95th percentile as well. WDYT?

Reply:
What are the 95th percentiles?

Reply:
Updated